CDCgov/prime-reportstream
# 13538: send original filename format report_id-original_name

This PR updates the name format for topics with `isSendOrginal=true`

Test Steps:
1. Execute
```
curl --location 'http://localhost:7071/api/reports' \
--header 'client: ignore.ignore-elr-elims' \
--header 'Content-Type: application/hl7-v2' \
--header 'x-functions-key: 09xifjf....444s' \
--header 'payloadname: external-name-test.hl7' \
--data 'MSH|^~\&|MEDITECH^2.16.840.1.114222.4.3.2.2.1.321.111^ISO|COCAA|DCEED.ELR^2.16.840.1.114222.4.1.144.2.5^ISO|CDPHE^2.16.840.1.114222.4.1.144^ISO|20280808152805+0000||ORU^R01^ORU_R01|MT_COCAA_ORU_AAPHELR.1.6214638|P|2.5.1
SFT|MEDITECH, Inc.^L^^^^MEDITECH&1.3.6.1.4.1.24310&ISO^XX^^^MEDITECH|5.67|Laboratory Applicati|20210810||20210810000000+0000
PID|1||100001^^^MEDITECH&2.16.840.1.114222.4.3.2.2.1.321.111&ISO^MR^COCAA~285-64-4175^^^MEDITECH&2.16.840.1.114222.4.3.2.2.1.321.111&ISO^SS^COCAA~X605236^^^MEDITECH&2.16.840.1.114222.4.3.2.2.1.321.111&ISO^PI^COCAA~E40014904640^^^MEDITECH&2.16.840.1.114222.4.3.2.2.1.321.111&ISO^AN^COCAA||^^^^^^L|||M||2106-3^https://www.hl7.org/fhir/v2/0005/^HL70005^W^White^L^2.5.1|^^^NM^^USA^H||^PRN^PH^^1^^^^^^^|^WPN^PH^^1^^^^Business^^^||S|NOE^Nonreligious^HL70006|||||N^https://www.hl7.org/fhir/v2/0189/^HL70189^2^Not Hispanic or Latin^L^2.5.1||||||||N|||20210809145200+0000|COCAA|337915000^Homo sapiens (organism)^SCT^Human^Human^L
ORC|RE|09339017^M12776123.1^2.16.840.1.114222.4.1.144^ISO||21:AA:B0029251S^M12776123^2.16.840.1.114222.4.1.144^ISO||N|||20280808092800+0000|||^^^^^^^^&2.16.840.1.113883.4.6&ISO^^^^NPI||^WPN^PH^^1^303^4362727^^Hospital Line^^^+1 303 436 2727|20210803235000+0000||TMCA, SOUTH CAMPUS||||TMCA, SOUTH CAMPUS^L^^^^TMCA, SOUTH CAMPUS&2.16.840.1.114222.4.1.144&ISO^XX^^^06D0055551|1501 S. POTOMAC^^AURORA^IG^80012^USA^B^^MD
OBR|1|09339017^M12776123.1^2.16.840.1.114222.4.1.144^ISO||600-7^Bacteria identified in Blood by Culture^LN^BCEXT^EXTENDED BLOOD CULTURE^L||||20280802085201+0000||ER|||SEPTIC SHOCK|||^^^^^^^^&2.16.840.1.113883.4.6&ISO^^^^NPI|^WPN^PH^^1^303^4362727^^Hospital Line^^^+1 303 436 2727|21:AA:B0029251Sm12776123|09339017||TMCA, SOUTH CAMPUS|20210810122500+0000|||P||||||R6521^SEVERE SEPSIS WITH SEPTIC SHOCK^I10^R6521^SEVERE SEPSIS WITH SEPTIC SHOCK^I10
OBX|1||600-7^Bacteria identified in Blood by Culture^LN^BCEXT^EXTENDED BLOOD CULTURE^L|1.1|||||||C|||20280802085201+0000|MCOE^ROSE MEDICAL CENTER (MCOE)^LB||||20280807223601+0000||||ROSE MEDICAL CENTER (MCOE)^L|4567 E. NINTH AVENUE^^DENVER^IG^80220^USA^B^^S|1033122247^Chahal^Shalini^^^Dr.^MD^^RML^L^^^FI^ROSE MEDICAL CENTER'
```
2. Verify that a report shows up in the sftp directory that looks like `<SOME_UUID>-external-name-test.hl7`

## Changes
- For `isSendOriginal=true` sets the file name to `{report.id}-{externalname}`

## Checklist

### Testing

### Process

## Linked Issues
- #13538 

## To Be Done

## Specific Security-related subjects a reviewer should pay specific attention to
